packages/subcommands/src/lib/Subcommand.ts

```diff
@@ -1,81 +1,89 @@
-		const group = interaction.options.getSubcommandGroup();
-
-		if (!subcommand && !group) {
-			throw new Error(`The command ${this.name} does not support sub-commands.`);
+		if (subCommandName && !subCommandGroupName) {
+			for (const mapping of this.subCommands) {
+				if (!(mapping instanceof ChatInputSubcommandMappings)) continue;
+				const subCommand = mapping.subcommands.find(({ name }) => name === subCommandName);
+				if (subCommand) return this.#handleInteractionRun(interaction, context, subCommand);
+			}
 		}
 
-		if (this.subCommands.has(subcommand) || this.subCommands.has(group)) {
-			const mappedSubcommand = this.subCommands.get(subcommand) ?? this.subCommands.get(group);
-
-			if (mappedSubcommand instanceof ChatInputSubcommandMappings) {
-				const toRun = mappedSubcommand.subcommands.find((scmd) => scmd.name === subcommand);
+		if (subCommandGroupName) {
+			for (const mapping of this.subCommands) {
+				if (!(mapping instanceof ChatInputSubcommandGroupMappings)) continue;
+				if (mapping.groupName !== subCommandGroupName) continue;
 
-				if (toRun) {
-					const result = await fromAsync(async () => {
-						interaction.client.emit(Events.SubcommandRun, { interaction, subcommand, ...context });
+				const subCommand = mapping.subcommands.find(({ name }) => name === subCommandName);
+				if (subCommand) return this.#handleInteractionRun(interaction, context, subCommand);
+			}
+		}
 
-						await toRun.to(interaction, context);
+		throw new Error(`The command ${this.name} does not support sub-commands.`);
+	}
 
-						interaction.client.emit(Events.SubcommandSuccess, { interaction, subcommand, ...context });
-					});
+	async #handleInteractionRun(interaction: CommandInteraction, context: ChatInputCommand.RunContext, subCommand: SubCommandMappingValue) {
+		const result = await fromAsync(async () => {
+			interaction.client.emit(Events.SubCommandMessageRun as never, interaction, subCommand, context);
+			await subCommand.to(interaction, context);
+			interaction.client.emit(Events.SubCommandMessageSuccess as never, interaction, subCommand.name, context);
+		});
 
-					if (isErr(result)) {
-						interaction.client.emit(Events.SubcommandError, result.error, {});
-					}
-				}
-			} else if (mappedSubcommand instanceof ChatInputSubcommandGroupMappings) {
-				// TODO: We have chat input group, run it and check result
-			}
+		if (isErr(result)) {
+			interaction.client.emit(Events.SubCommandMessageSuccess as never, result.error, context);
 		}
 	}
 
-	#handleChatinput() {}
+	async #handleMessageRun(message: Message, args: Args, context: MessageCommandContext, subCommand: SubCommandMessageRunMappingValue) {
+		const result = await fromAsync(async () => {
+			message.client.emit(Events.SubCommandMessageRun as never, message, subCommand, context);
+			await subCommand.to(message, args, context);
+			message.client.emit(Events.SubCommandMessageSuccess as never, message, subCommand.name, context);
```

Comment:
Why is `subCommand.name` on the message success but not the message run?

looks out of place, but I don't know how it was originally made